### DISCUSSION TITLE
Android uianimator2 ignoring noReset:true fullReset:false

### DISCUSSION THREAD

**Discussion by sidferreira** [appium/appium#17370] [Q&A] [answered]
```
appium@1.22.3
appium-android-driver@4.20.0
appium-uiautomator2-driver@1.75.0
```

I'm using WebDriverIO with Appium for React Native, and it seems that appium is ignoring the `noReset:true, fullReset:false` capabilities. I'm running the server manually, and it seems that as soon as the session is deleted, the app is killed.

There's no explicit message on the log saying it is resetting, closing the app, or anything.

Any suggestions on how to debug it?

---

Capabilities received by appium:

```
[debug] [BaseDriver] Creating session with W3C capabilities: {
[debug] [BaseDriver]   "alwaysMatch": {
[debug] [BaseDriver]     "platformName": "Android",
[debug] [BaseDriver]     "browserName": "",
[debug] [BaseDriver]     "wdio:sharedStoreServicePort": 50247,
[debug] [BaseDriver]     "appium:platformVersion": "12.0",
[debug] [BaseDriver]     "appium:app": "...",
[debug] [BaseDriver]     "appium:appName": "...",
[debug] [BaseDriver]     "appium:automationName": "UiAutomator2",
[debug] [BaseDriver]     "appium:noReset": true,
[debug] [BaseDriver]     "appium:fullReset": false,
[debug] [BaseDriver]     "appium:eventTimings": true,
[debug] [BaseDriver]     "appium:language": "en",
[debug] [BaseDriver]     "appium:locale": "US"
[debug] [BaseDriver]   },
[debug] [BaseDriver]   "firstMatch": [
[debug] [BaseDriver]     {}
[debug] [BaseDriver]   ]
[debug] [BaseDriver] }
```

**Comment by RSM-SBA** [comment]
I think that is normal behavior. Why you are closing the session, if you still want to test the app ?
If you want to restart the app you should use terminateApp and mobile:LaunchApp (for IOS)

**Reply by sidferreira** [reply]
> I think that is normal behavior. Why you are closing the session, if you still want to test the app ?

@StephanB75 on iOS it doesn't happen like that, and I believe that webdriver is the one actually closing/deleting the session.

Do you mean the flags have no use?

**Reply by RSM-SBA** [reply]
Maybe i'am wrong but "noreset" says "Do not destroy or shut down sim after test."

**Reply by sidferreira** [reply]
@StephanB75 according to https://appium.io/docs/en/writing-running-appium/other/reset-strategies/ on android noreset:
`Do not stop app, do not clear app data, and do not uninstall apk.`

So I assume it shouldn't close the app...

**Reply by sidferreira** [reply]
It is actually doing the default behavior...

**Reply by sidferreira** [reply]
@StephanB75 I just found a `dontStopAppOnReset` flag that actually does the trick... I can't understand why it exists...

**Reply by jlipps** [reply]
yeah the default `noReset` behaviour still stops the app, it just doesn't reset/clear any data. you found the correct capability to add not stopping as well.

**Comment by sidferreira** [comment, marked as answer]
Using `dontStopAppOnReset` did the trick...
